### ISSUE TITLE
Use a (code style) linter

### ISSUE THREAD

**Issue by wb9688** [TeamNewPipe/NewPipe#3223]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [x] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

It would be nice if we would use a (code style) linter, so we don't have to manually point out missing spaces, etc. It should be run in the CI. This is not only relevant for NewPipe, but also NewPipeExtractor and PipeCast.

**Comment by opusforlife2** [comment]
Has this been done with CheckStyle or is that something else entirely?

**Comment by wb9688** [comment]
@opusforlife2: Yes, but I want to change the way we handle suppressions and have it for NewPipeExtractor as well
